### PR TITLE
Split ration and inconsistency removal for ensemble

### PR DESCRIPTION
**Pull request by sfluegel05**
To use the chebai-models for the [chebifier](https://github.com/ChEB-AI/python-chebifier)-ensemble, I have made some minor changes:
- Split ratios can be set independently for validation and test set - this allows e.g. for a larger validation set which we need to calculate metrics or (potentially) do some ensemble-training
- Prediction smoothing is now more general - we use this in the ensemble to ensure all predictions are consistent

### CODE REVIEW

**Comment by aditya0by0** [comment]
@sfluegel05,

**Before this PR**, with the default values,

* **Test split:** 12.75% of the data
* **Validation split:** \~2.309% of the data
* **Train split:** \~84.94% of the data

**After this PR**, with the new default values,

* **Test split:** 10% of the data
* **Validation split:** \~5.004% of the data
* **Train split:** \~84.9999% of the data

Shouldn’t we adjust the new default values to maintain the same data distribution as before the change?

**Comment by sfluegel05** [comment]
Well, changing the default values kind of was my intention. I don't think there has ever been an actual reason for the validation set being 15% of 15%. I can't think of a scenario where it is important to have the same split sizes but where you are not re-using the actual splits.
The advantage of the new ratio is that you get more validation data (for ChEBI50, the smallest classes have only 50 members, meaning 2.3% equates to 1 positive sample in the validation set, which is not that expressive).

**Comment by sfluegel05** [comment]
Thanks, that is my bad. Somehow I missed that the tests were failing for this PR.
